csrc/multidevice/executor.cpp

```diff
@@ -155,12 +155,13 @@ void MultiDeviceExecutor::postKernel(SegmentedGroup* group) {
 
   // Compile the group and execute it with FusionExecutor
   // Check if the executor has been cached. If not, create and cache it
-  if (fe_.find(group) == fe_.end()) {
-    fe_.emplace(group, std::make_unique<FusionExecutor>());
-    fusions_.emplace(group, staged_fusion_->makeFusion(group));
-    fe_[group]->compileFusion(fusions_.at(group).get(), group_input_IValues);
+  if (fec_.find(group) == fec_.end()) {
+    fec_.emplace(
+        group,
+        std::make_unique<FusionExecutorCache>(
+            staged_fusion_->makeFusion(group), 0, auto_schedule_));
```

Comment:
Can you clarify the reason for disabling auto scheduling? I guess: because of multi-level segmentation, `staged_fusion` may have to be further segmented before it can be auto-scheduled.

Reply:
We disable auto-scheduling because there is only manual scheduling available for multi-device, which is not compatible with auto-scheduling (I tried to clarify this is in the PR description)

Reply:
Question to @naoyam for my education: regardless of multi-device, how does FusionExecutorCache behave when the input fusion is already (partially) scheduled (e.g. using `tv2->split`)? Does it overwrite those schedules or try to preserve them?

Reply:
It assumes no scheduling is done, so I'm actually a bit surprised it doesn't fail 😆

Reply:
Now, there is no check that the fusion is not pre-scheduled, it is only implicitely assumed. The scheduler doesn't try to preserve pre-scheduling, it will just blindly try to apply scheduling, and might stumble by mistake on an assert error such as [this one](https://github.com/NVIDIA/Fuser/blob/71d9b555b401655c0d664b94e8abb7021ce29637/csrc/tensor_view.cpp#L665).